OsmAndMapCreator/src/net/osmand/BinaryMerger.java

```diff
@@ -286,6 +302,71 @@ private void combineAddressIndex(String name, BinaryMapIndexWriter writer, Addre
 		writer.endWriteAddressIndex();
 	}
 
+	private void combinePoiIndex(String name, BinaryMapIndexWriter writer, long dateCreated, PoiRegion[] poiRegions, BinaryMapIndexReader[] indexes)
+			throws IOException, SQLException {
+		int readPoiCount = 0;
+		int badNeg = 0;
+//		int writtenPoiCount = 0;
+		MapRenderingTypesEncoder renderingTypes = new MapRenderingTypesEncoder(null, name);
+		boolean overwriteIds = true;
+		IndexPoiCreator indexPoiCreator = new IndexPoiCreator(renderingTypes, overwriteIds);
+		indexPoiCreator.createDatabaseStructure(new File(new File(System.getProperty("user.dir")), IndexCreator.getPoiFileName(name)));
+		List<Long> ids = new ArrayList<Long>();
+		Amenity a = null;
+		for (int i = 0; i < poiRegions.length; i++) {
+			BinaryMapIndexReader index = indexes[i];
+			log.info("Region: " + extractRegionName(index));
+			List<Amenity> amenities = index.searchPoi(BinaryMapIndexReader.buildSearchPoiRequest(
+					MapUtils.MAP_BOUNDING_BOX_31_TILE_NUMBER[0],
+					MapUtils.MAP_BOUNDING_BOX_31_TILE_NUMBER[1],
+					MapUtils.MAP_BOUNDING_BOX_31_TILE_NUMBER[2],
+					MapUtils.MAP_BOUNDING_BOX_31_TILE_NUMBER[3],
+					MapUtils.NO_ZOOM,
+					BinaryMapIndexReader.EMPTY_SEARCH_POI_TYPE_FILTER,
+					null));
+			for (Amenity amenity : amenities) {
+				readPoiCount++;
+				boolean isNeg = amenity.getId() < 0;
+				IndexPoiCreator.updateId(amenity);
+				if (isNeg && amenity.getId() > 0) {
+					badNeg++;
+				}
+				ids.add(amenity.getId());
+				boolean isWritten = indexPoiCreator.insertAmenityIntoPoi(amenity);
+				if (Arrays.asList(8316643308L, 8316171782L, 8316171780L).contains(amenity.getId())) {
```

Comment:
Would be good to use debug or may be constants to avoid these in the final code.